app/controllers/offers_controller.rb

```diff
@@ -1,12 +1,11 @@
 class OffersController < ApplicationController
   def create
     @shift = Shift.find(params[:shift_id])
-    @user = current_user
-    Offer.where("offerer_id=? AND shift_id=?", @user.id, @shift.id).delete_all
+    Offer.where("offerer_id=? AND shift_id=?", current_user.id, @shift.id).delete_all
```

Comment:
You could also write this line like

```ruby
Offer.where(offerer_id: current_user.id, shift_id: @shift.id).delete_all
```

Or 

```ruby
@shift.offers.where(offerer_id: current_user.id).delete_all
```

This last option has the benefit of being shortest and (in my opinion) showing the clearest intent (we are "deleting the current users offers for this shift").